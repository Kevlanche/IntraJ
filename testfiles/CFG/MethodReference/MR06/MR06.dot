digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "InterfaceDecl[5]"; "ClassDecl[22]"; "ClassDecl[51]"}
{ rank = same; "Modifiers[6]"; "List[8]"; "List[9]"; "Modifiers[23]"; "Opt[25]"; "List[26]"; "List[27]"; "Modifiers[52]"; "Opt[55]"; "List[56]"; "List[57]"}
{ rank = same; "List[7]"; "MethodDecl[10]"; "List[24]"; "ConstructorDecl[28]"; "List[53]"; "MethodDecl[58]"}
{ rank = same; "Entry[11]"; "Modifiers[12]"; "TypeAccess[14]"; "List[15]"; "List[20]"; "Opt[21]"; "Exit[22]"; "Entry[29]"; "Implicit[29]"; "Modifiers[29]"; "List[31]"; "List[36]"; "Opt[37]"; "Block[38]"; "Exit[29]"; "ExprStmt[51]"; "Modifier[54]"; "Entry[59]"; "Modifiers[60]"; "PrimitiveTypeAccess[64]"; "List[65]"; "List[71]"; "Opt[72]"; "Exit[73]"}
{ rank = same; "List[13]"; "ParameterDeclaration[16]"; "List[30]"; "ParameterDeclaration[32]"; "List[39]"; "SuperConstructorAccess[52]"; "List[61]"; "ParameterDeclaration[66]"; "Block[73]"}
{ rank = same; "Modifiers[17]"; "TypeAccess[19]"; "Modifiers[33]"; "TypeAccess[35]"; "ExprStmt[40]"; "List[53]"; "Modifier[62]"; "Modifier[63]"; "Modifiers[67]"; "ArrayTypeAccess[69]"; "List[74]"}
{ rank = same; "List[18]"; "List[34]"; "Dot[41]"; "List[68]"; "TypeAccess[70]"; "VarDeclStmt[75]"; "ExprStmt[86]"}
{ rank = same; "Dot[42]"; "MethodAccess[45]"; "Modifiers[76]"; "TypeAccess[78]"; "List[79]"; "Dot[87]"}
{ rank = same; "TypeAccess[43]"; "VarAccess[44]"; "List[47]"; "List[49]"; "List[50]"; "UncheckedExceptions[46]"; "List[77]"; "VariableDeclarator[80]"; "VarAccess[88]"; "MethodAccess[89]"}
{ rank = same; "VarAccess[48]"; "List[81]"; "Opt[82]"; "List[91]"; "List[93]"; "List[94]"; "UncheckedExceptions[90]"}
{ rank = same; "ClassReference[83]"; "StringLiteral[92]"}
{ rank = same; "TypeAccess[84]"; "List[85]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InterfaceDecl[5]" [label="InterfaceDecl\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[8]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[9]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[10]" [label="MethodDecl\n < getMessage >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[11]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[12]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[13]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[14]" [label="TypeAccess\n < Message >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[16]" [label="ParameterDeclaration\n < msg >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[17]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[18]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[19]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[20]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[21]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[22]" [label="Exit\n ", fillcolor=white   style=filled  ]
"ClassDecl[22]" [label="ClassDecl\n < Message >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[23]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[24]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstructorDecl[28]" [label="ConstructorDecl\n < Message >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[29]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Implicit[29]" [label="Implicit\n < Instance >", fillcolor=white   style=filled  ]
"Modifiers[29]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[31]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[32]" [label="ParameterDeclaration\n < msg >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[33]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[35]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[36]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[37]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[38]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[39]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[40]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[41]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[42]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[43]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[44]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[45]" [label="MethodAccess\n < print >", fillcolor=white   style=filled  ]
"List[47]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[48]" [label="VarAccess\n < msg >", fillcolor=white   style=filled  ]
"List[49]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[50]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[46]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[29]" [label="Exit\n ", fillcolor=white   style=filled  ]
"ExprStmt[51]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"SuperConstructorAccess[52]" [label="SuperConstructorAccess\n < super >", fillcolor=white   style=filled  ]
"List[53]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[51]" [label="ClassDecl\n < MR06 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[52]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[53]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[54]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[55]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[56]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[57]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[58]" [label="MethodDecl\n < main >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[59]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[60]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[61]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[62]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[63]" [label="Modifier\n < static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[64]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[65]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[66]" [label="ParameterDeclaration\n < args >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[67]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[68]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ArrayTypeAccess[69]" [label="ArrayTypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[70]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[71]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[72]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[73]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[74]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[75]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[76]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[77]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[78]" [label="TypeAccess\n < Messageable >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[79]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[80]" [label="VariableDeclarator\n < Messageable foo >", fillcolor=white   style=filled  ]
"List[81]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[82]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassReference[83]" [label="ClassReference\n < Message::<>new >", fillcolor=white   style=filled  ]
"TypeAccess[84]" [label="TypeAccess\n < Message >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[85]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[86]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[87]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[88]" [label="VarAccess\n < foo >", fillcolor=white   style=filled  ]
"MethodAccess[89]" [label="MethodAccess\n < getMessage >", fillcolor=white   style=filled  ]
"List[91]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[92]" [label="StringLiteral\n < MR06 >", fillcolor=white   style=filled  ]
"List[93]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[94]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[90]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[73]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"InterfaceDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"InterfaceDecl[5]" -> "List[8]" [style=dashed, color=gray]
"MethodDecl[10]" -> "Entry[11]" [style=dashed, color=gray]
"Modifiers[12]" -> "List[13]" [style=dashed, color=gray]
"MethodDecl[10]" -> "Modifiers[12]" [style=dashed, color=gray]
"MethodDecl[10]" -> "TypeAccess[14]" [style=dashed, color=gray]
"Modifiers[17]" -> "List[18]" [style=dashed, color=gray]
"ParameterDeclaration[16]" -> "Modifiers[17]" [style=dashed, color=gray]
"ParameterDeclaration[16]" -> "TypeAccess[19]" [style=dashed, color=gray]
"List[15]" -> "ParameterDeclaration[16]" [style=dashed, color=gray]
"MethodDecl[10]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[10]" -> "List[20]" [style=dashed, color=gray]
"MethodDecl[10]" -> "Opt[21]" [style=dashed, color=gray]
"MethodDecl[10]" -> "Exit[22]" [style=dashed, color=gray]
"List[9]" -> "MethodDecl[10]" [style=dashed, color=gray]
"InterfaceDecl[5]" -> "List[9]" [style=dashed, color=gray]
"List[4]" -> "InterfaceDecl[5]" [style=dashed, color=gray]
"Modifiers[23]" -> "List[24]" [style=dashed, color=gray]
"ClassDecl[22]" -> "Modifiers[23]" [style=dashed, color=gray]
"ClassDecl[22]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[22]" -> "List[26]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Entry[29]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Implicit[29]" [style=dashed, color=gray]
"Modifiers[29]" -> "List[30]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Modifiers[29]" [style=dashed, color=gray]
"Modifiers[33]" -> "List[34]" [style=dashed, color=gray]
"ParameterDeclaration[32]" -> "Modifiers[33]" [style=dashed, color=gray]
"ParameterDeclaration[32]" -> "TypeAccess[35]" [style=dashed, color=gray]
"List[31]" -> "ParameterDeclaration[32]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "List[31]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "List[36]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Opt[37]" [style=dashed, color=gray]
"Dot[42]" -> "TypeAccess[43]" [style=dashed, color=gray]
"Dot[42]" -> "VarAccess[44]" [style=dashed, color=gray]
"Dot[41]" -> "Dot[42]" [style=dashed, color=gray]
"List[47]" -> "VarAccess[48]" [style=dashed, color=gray]
"MethodAccess[45]" -> "List[47]" [style=dashed, color=gray]
"MethodAccess[45]" -> "List[49]" [style=dashed, color=gray]
"MethodAccess[45]" -> "List[50]" [style=dashed, color=gray]
"MethodAccess[45]" -> "UncheckedExceptions[46]" [style=dashed, color=gray]
"Dot[41]" -> "MethodAccess[45]" [style=dashed, color=gray]
"ExprStmt[40]" -> "Dot[41]" [style=dashed, color=gray]
"List[39]" -> "ExprStmt[40]" [style=dashed, color=gray]
"Block[38]" -> "List[39]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Block[38]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Exit[29]" [style=dashed, color=gray]
"SuperConstructorAccess[52]" -> "List[53]" [style=dashed, color=gray]
"ExprStmt[51]" -> "SuperConstructorAccess[52]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "ExprStmt[51]" [style=dashed, color=gray]
"List[27]" -> "ConstructorDecl[28]" [style=dashed, color=gray]
"ClassDecl[22]" -> "List[27]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[22]" [style=dashed, color=gray]
"List[53]" -> "Modifier[54]" [style=dashed, color=gray]
"Modifiers[52]" -> "List[53]" [style=dashed, color=gray]
"ClassDecl[51]" -> "Modifiers[52]" [style=dashed, color=gray]
"ClassDecl[51]" -> "Opt[55]" [style=dashed, color=gray]
"ClassDecl[51]" -> "List[56]" [style=dashed, color=gray]
"MethodDecl[58]" -> "Entry[59]" [style=dashed, color=gray]
"List[61]" -> "Modifier[62]" [style=dashed, color=gray]
"List[61]" -> "Modifier[63]" [style=dashed, color=gray]
"Modifiers[60]" -> "List[61]" [style=dashed, color=gray]
"MethodDecl[58]" -> "Modifiers[60]" [style=dashed, color=gray]
"MethodDecl[58]" -> "PrimitiveTypeAccess[64]" [style=dashed, color=gray]
"Modifiers[67]" -> "List[68]" [style=dashed, color=gray]
"ParameterDeclaration[66]" -> "Modifiers[67]" [style=dashed, color=gray]
"ArrayTypeAccess[69]" -> "TypeAccess[70]" [style=dashed, color=gray]
"ParameterDeclaration[66]" -> "ArrayTypeAccess[69]" [style=dashed, color=gray]
"List[65]" -> "ParameterDeclaration[66]" [style=dashed, color=gray]
"MethodDecl[58]" -> "List[65]" [style=dashed, color=gray]
"MethodDecl[58]" -> "List[71]" [style=dashed, color=gray]
"Modifiers[76]" -> "List[77]" [style=dashed, color=gray]
"VarDeclStmt[75]" -> "Modifiers[76]" [style=dashed, color=gray]
"VarDeclStmt[75]" -> "TypeAccess[78]" [style=dashed, color=gray]
"VariableDeclarator[80]" -> "List[81]" [style=dashed, color=gray]
"ClassReference[83]" -> "TypeAccess[84]" [style=dashed, color=gray]
"ClassReference[83]" -> "List[85]" [style=dashed, color=gray]
"Opt[82]" -> "ClassReference[83]" [style=dashed, color=gray]
"VariableDeclarator[80]" -> "Opt[82]" [style=dashed, color=gray]
"List[79]" -> "VariableDeclarator[80]" [style=dashed, color=gray]
"VarDeclStmt[75]" -> "List[79]" [style=dashed, color=gray]
"List[74]" -> "VarDeclStmt[75]" [style=dashed, color=gray]
"Dot[87]" -> "VarAccess[88]" [style=dashed, color=gray]
"List[91]" -> "StringLiteral[92]" [style=dashed, color=gray]
"MethodAccess[89]" -> "List[91]" [style=dashed, color=gray]
"MethodAccess[89]" -> "List[93]" [style=dashed, color=gray]
"MethodAccess[89]" -> "List[94]" [style=dashed, color=gray]
"MethodAccess[89]" -> "UncheckedExceptions[90]" [style=dashed, color=gray]
"Dot[87]" -> "MethodAccess[89]" [style=dashed, color=gray]
"ExprStmt[86]" -> "Dot[87]" [style=dashed, color=gray]
"List[74]" -> "ExprStmt[86]" [style=dashed, color=gray]
"Block[73]" -> "List[74]" [style=dashed, color=gray]
"Opt[72]" -> "Block[73]" [style=dashed, color=gray]
"MethodDecl[58]" -> "Opt[72]" [style=dashed, color=gray]
"MethodDecl[58]" -> "Exit[73]" [style=dashed, color=gray]
"List[57]" -> "MethodDecl[58]" [style=dashed, color=gray]
"ClassDecl[51]" -> "List[57]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[51]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[22]" -> "Entry[11]" [color=red, constraint=false]
"Entry[11]" -> "Exit[22]" [color=blue, constraint=false, xlabel=" "] 
"Exit[29]" -> "MethodAccess[45]" [color=red, constraint=false]
"MethodAccess[45]" -> "Exit[29]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[45]" -> "VarAccess[48]" [color=red, constraint=false]
"VarAccess[48]" -> "MethodAccess[45]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[48]" -> "VarAccess[44]" [color=red, constraint=false]
"VarAccess[44]" -> "VarAccess[48]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[44]" -> "TypeAccess[43]" [color=red, constraint=false]
"TypeAccess[43]" -> "VarAccess[44]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[43]" -> "Implicit[29]" [color=red, constraint=false]
"Implicit[29]" -> "TypeAccess[43]" [color=blue, constraint=false, xlabel=" "] 
"Implicit[29]" -> "SuperConstructorAccess[52]" [color=red, constraint=false]
"SuperConstructorAccess[52]" -> "Implicit[29]" [color=blue, constraint=false, xlabel=" "] 
"SuperConstructorAccess[52]" -> "Entry[29]" [color=red, constraint=false]
"Entry[29]" -> "SuperConstructorAccess[52]" [color=blue, constraint=false, xlabel=" "] 
"Exit[73]" -> "MethodAccess[89]" [color=red, constraint=false]
"MethodAccess[89]" -> "Exit[73]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[89]" -> "StringLiteral[92]" [color=red, constraint=false]
"StringLiteral[92]" -> "MethodAccess[89]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[92]" -> "VarAccess[88]" [color=red, constraint=false]
"VarAccess[88]" -> "StringLiteral[92]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[88]" -> "VariableDeclarator[80]" [color=red, constraint=false]
"VariableDeclarator[80]" -> "VarAccess[88]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[80]" -> "ClassReference[83]" [color=red, constraint=false]
"ClassReference[83]" -> "VariableDeclarator[80]" [color=blue, constraint=false, xlabel=" "] 
"ClassReference[83]" -> "Entry[59]" [color=red, constraint=false]
"Entry[59]" -> "ClassReference[83]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InterfaceDecl[5]" -> "ClassDecl[22]" -> "ClassDecl[51]" [style=invis]
"Modifiers[6]" -> "List[8]" -> "List[9]" [style=invis]
"Entry[11]" -> "Modifiers[12]" -> "TypeAccess[14]" -> "List[15]" -> "List[20]" -> "Opt[21]" -> "Exit[22]" [style=invis]
"Modifiers[17]" -> "TypeAccess[19]" [style=invis]
"Modifiers[23]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"Entry[29]" -> "Modifiers[29]" -> "List[31]" -> "List[36]" -> "Opt[37]" -> "Block[38]" -> "Exit[29]" [style=invis]
"Modifiers[33]" -> "TypeAccess[35]" [style=invis]
"Dot[42]" -> "MethodAccess[45]" [style=invis]
"TypeAccess[43]" -> "VarAccess[44]" [style=invis]
"Modifiers[52]" -> "Opt[55]" -> "List[56]" -> "List[57]" [style=invis]
"Entry[59]" -> "Modifiers[60]" -> "PrimitiveTypeAccess[64]" -> "List[65]" -> "List[71]" -> "Opt[72]" -> "Exit[73]" [style=invis]
"Modifier[62]" -> "Modifier[63]" [style=invis]
"Modifiers[67]" -> "ArrayTypeAccess[69]" [style=invis]
"VarDeclStmt[75]" -> "ExprStmt[86]" [style=invis]
"Modifiers[76]" -> "TypeAccess[78]" -> "List[79]" [style=invis]
"List[81]" -> "Opt[82]" [style=invis]
"TypeAccess[84]" -> "List[85]" [style=invis]
"VarAccess[88]" -> "MethodAccess[89]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
103", fillcolor=1]
Edges [label="Total Nr. Edges
130", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
15", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
14", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}