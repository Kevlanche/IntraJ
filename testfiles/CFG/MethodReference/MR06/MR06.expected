NodeName: Entry[11] Successors: [Exit[22]]
NodeName: Exit[22] Successors: []
NodeName: Entry[29] Successors: [SuperConstructorAccess[52]]
NodeName: SuperConstructorAccess[52] Successors: [Implicit[29]]
NodeName: Implicit[29] Successors: [TypeAccess[43]]
NodeName: TypeAccess[43] Successors: [VarAccess[44]]
NodeName: VarAccess[44] Successors: [VarAccess[48]]
NodeName: VarAccess[48] Successors: [MethodAccess[45]]
NodeName: MethodAccess[45] Successors: [Exit[29]]
NodeName: Exit[29] Successors: []
NodeName: Entry[59] Successors: [ClassReference[83]]
NodeName: ClassReference[83] Successors: [VariableDeclarator[80]]
NodeName: VariableDeclarator[80] Successors: [VarAccess[88]]
NodeName: VarAccess[88] Successors: [StringLiteral[92]]
NodeName: StringLiteral[92] Successors: [MethodAccess[89]]
NodeName: MethodAccess[89] Successors: [Exit[73]]
NodeName: Exit[73] Successors: []

NodeName: Exit[22] Predecessor: [Entry[11]]
NodeName: Entry[11] Predecessor: []
NodeName: Exit[29] Predecessor: [MethodAccess[45]]
NodeName: MethodAccess[45] Predecessor: [VarAccess[48]]
NodeName: VarAccess[48] Predecessor: [VarAccess[44]]
NodeName: VarAccess[44] Predecessor: [TypeAccess[43]]
NodeName: TypeAccess[43] Predecessor: [Implicit[29]]
NodeName: Implicit[29] Predecessor: [SuperConstructorAccess[52]]
NodeName: SuperConstructorAccess[52] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: Exit[73] Predecessor: [MethodAccess[89]]
NodeName: MethodAccess[89] Predecessor: [StringLiteral[92]]
NodeName: StringLiteral[92] Predecessor: [VarAccess[88]]
NodeName: VarAccess[88] Predecessor: [VariableDeclarator[80]]
NodeName: VariableDeclarator[80] Predecessor: [ClassReference[83]]
NodeName: ClassReference[83] Predecessor: [Entry[59]]
NodeName: Entry[59] Predecessor: []
